pkg/controller/gitopscluster/gitopscluster_controller.go

```diff
@@ -184,7 +184,7 @@ func (r *ReconcileGitOpsCluster) Reconcile(ctx context.Context, request reconcil
 	gitOpsClusters, err := r.GetAllGitOpsClusters()
 
 	if err != nil {
-		return reconcile.Result{Requeue: false}, nil
+		return reconcile.Result{Requeue: false}, err
```

Comment:
This is a concerning change? Why is it necessary is it a fix that we did and we forgot to put it in the main branch?

/hold